src/db/serde.ts

```diff
@@ -0,0 +1,98 @@
+import { Column, Insertable, Table } from "zapatos/schema";
+
+export interface Hook<U, T> {
+  [t: Table]: { [c: Column]: (x: U) => T };
+}
+
+// TODO: narrow these types
+const DESERIALIZE_HOOK: Hook<any, any> = {};
+const SERIALIZE_HOOK: Hook<any, any> = {};
+
+function applyHook<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable | Insertable[]
+): Insertable | Insertable[] {
+  return Array.isArray(values)
+    ? values.map((v) => applyHookSingle(hook, table, v))
+    : applyHookSingle(hook, table, values);
+}
+
+function applyHookSingle<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable
```

Comment:
I think we can use InsertableForTable types...I guess that's an overarching todo, to see if we can leverage the generated types to get a bit more type safety